### PR TITLE
AWS OIDC: List EKS Clusters service

### PR DESCRIPTION
**Pull request by marcoandredinis**
This PR creates a new method on the AWS OIDC gRPC service that lists eks clusters.

This is part of a refactor that moves the API calls behind the Auth Service.

Context: https://github.com/gravitational/teleport/issues/37245

### CODE REVIEW

File: api/proto/teleport/integration/v1/awsoidc_service.proto [hunk 1]
```diff
@@ -257,3 +263,42 @@ message ListEC2Response {
...
+// EKSCluster represents an Amazon EKS Cluster.
+message EKSCluster {
+  // Name is the name of AWS EKS cluster.
+  string name = 1;
+  // Region is an AWS region.
+  string region = 2;
+  // Arn is an AWS ARN identification of the EKS cluster.
+  string arn = 3;
+  // Labels are labels of a EKS cluster.
+  map<string, string> labels = 4;
```
Comment (tigrato) [inline api/proto/teleport/integration/v1/awsoidc_service.proto:289]: by labels you mean AWS tags, is that correct? 
If yes, tags support null values

Comment (marcoandredinis) [reply api/proto/teleport/integration/v1/awsoidc_service.proto:289]: I would expect the converter to have an empty string.
However, this ends up being the same as is exposed in the SDK:
https://github.com/aws/aws-sdk-go-v2/blob/7888f0e10eb7ef5db0b67a8a27bb8eeee984257b/service/eks/types/types.go#L356-L359

Also, this is mostly exposing the internal API as a gRPC service, and this section was mostly a copy-paste from https://github.com/gravitational/teleport/blob/943bef63d92bbd06c0ca2ad018f2e25d5a80a4a2/lib/integrations/awsoidc/eks_list_clusters.go#L66-L67

File: api/proto/teleport/integration/v1/awsoidc_service.proto [hunk 2]
```diff
@@ -49,6 +49,12 @@ service AWSOIDCService {
   // It uses the following API:
   // https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstances.html
   rpc ListEC2(ListEC2Request) returns (ListEC2Response);
+
+  // ListEKSClusters lists the EKS Cluster of the AWS account per region.
+  // It uses the following APIs:
```
Comment (tigrato) [inline api/proto/teleport/integration/v1/awsoidc_service.proto:54]:
```suggestion
  // ListEKSClusters retrieves a paginated list of EKS clusters in the specified AWS region for a specific account.
```

File: api/proto/teleport/integration/v1/awsoidc_service.proto [hunk 3]
```diff
@@ -307,3 +313,42 @@ message ListEC2Response {
...
+  string region = 2;
+  // Arn is an AWS ARN identification of the EKS cluster.
+  string arn = 3;
+  // Labels are labels of a EKS cluster.
+  map<string, string> labels = 4;
+  // JoinLabels are Teleport labels that should be injected into kube agent
+  // if the cluster will be enrolled into Teleport (agent installed on it).
+  map<string, string> join_labels = 5;
+  // Status is a current status of an EKS cluster in AWS.
+  string status = 6;
```
Comment (tigrato) [inline api/proto/teleport/integration/v1/awsoidc_service.proto:344]: status is a enum, can we make it a enum or identify which possible values are we receiving

**Review by tigrato — APPROVED** [review]

**Review by AntonAM — APPROVED** [review]

**Pull request merged**